src/kestrel_datasource_stixshifter/query.py

```diff
@@ -113,10 +121,15 @@ def query_datasource(uri, pattern, session_id, config, store):
                 config["options"]["translation_workers_count"],
                 dsl["queries"],
                 raw_records_queue,
+                profile_limit,
             ):
                 for _ in range(config["options"]["translation_workers_count"]):
                     for packet in iter(translated_data_queue.get, STOP_SIGN):
                         if packet.success:
+                            num_objects = len(packet.data.get("objects", []))
```

Comment:
`packet` can either be STIX bundle or DataFrame, depending on the option in translation. We may need to handle both cases. Check the `ingest()` function to better see the formats.

I suggest to put this section into a function like `ingest()` when it gets larger with two options.

Reply:
Fixed in https://github.com/opencybersecurityalliance/kestrel-lang/pull/365/commits/6be265cf4490884e7dcc703ac2aee4a29a936db1